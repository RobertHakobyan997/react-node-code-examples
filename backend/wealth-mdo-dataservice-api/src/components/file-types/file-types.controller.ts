import { Controller, Get } from '@nestjs/common';
import { ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
import { Log } from 'ngpd-merceros-wealth-mdo-common-be/dist/logger/mdo-logger';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';

import { FileTypesService } from './file-types.service';

@ApiTags('File Types')
@Controller('file-types')
export class FileTypesController {
  constructor(private readonly service: FileTypesService) {
  }

  @ApiOperation({
    description: 'Get all file types',
  })
  @ApiOkResponse({
    description: 'Returns all file types',
    type       : [ Enum ],
  })
  @Get()
  @Log('Get supported files types')
  findAll() {
    return this.service.findAll();
  }
}
