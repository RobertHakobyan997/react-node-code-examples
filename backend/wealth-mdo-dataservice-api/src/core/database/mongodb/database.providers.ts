import mongoose from 'mongoose';
import { Logger } from '@nestjs/common';
import { IS_LOCAL } from 'ngpd-merceros-wealth-mdo-common-be/dist/utils/environments';

import mongoConfig from '../../../config/mongo.config';

export const databaseProviders = [
  {
    // ToDo: change to dynamic provider and take it from mongoConfig
    provide   : 'DatabaseConnection',
    useFactory: (): Promise<typeof mongoose> => {
      if (IS_LOCAL()) {
        mongoose.set('debug', (collectionName, methodName, ...methodArgs) => {
          const logger = new Logger(collectionName);
          const args = methodArgs.reduce((acc, next, idx) => {
            acc += JSON.stringify(next, null, 2) + (idx === methodArgs.length - 1 ? '' : ', ');
            return acc;
          }, '');
          logger.log(`${methodName}(${args})`);
        });
      }
      return mongoose.connect(mongoConfig().uri, mongoConfig().options);
    },
  },
];
