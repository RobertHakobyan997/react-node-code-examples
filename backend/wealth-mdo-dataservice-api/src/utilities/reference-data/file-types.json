[
  {
    "_id": "ObjectId('6075ffd013a048976c4c26bf')",
    "key": "positions",
    "display": "Positions",
    "order": 1,
    "hidden": false
  },
  {
    "_id": "ObjectId('6075ffd013a048976c4c26c0')",
    "key": "transactions",
    "display": "Transactions",
    "order": 2,
    "hidden": false
  },
  {
    "_id": "ObjectId('6075ffd013a048976c4c26c1')",
    "key": "cashFlow",
    "display": "Cash Flow",
    "order": 3,
    "hidden": false
  },
  {
    "_id": "ObjectId('6075ffd013a048976c4c26c2')",
    "key": "error",
    "display": "Error",
    "order": 4,
    "hidden": true
  },
  {
    "_id": "ObjectId('6075ffd013a048976c4c26cf')",
    "key": "prices",
    "display": "Prices",
    "order": 5,
    "hidden": false
  },
  {
    "_id": "ObjectId('6075ffd013a048976c4c27cf')",
    "key": "performance",
    "display": "Performance",
    "order": 6,
    "hidden": false
  },
  {
    "_id": "ObjectId('61c9c5a8557d7555fe297218')",
    "key": "various",
    "display": "Various",
    "order": 7,
    "hidden": false
  },
  {
    "_id": "ObjectId('621387e2a37c0841868f5bb2')",
    "key": "fxRates",
    "display": "FX Rates",
    "order": 8,
    "hidden": false
  }
]
