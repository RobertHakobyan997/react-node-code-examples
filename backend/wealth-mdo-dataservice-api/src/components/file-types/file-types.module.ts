import { Module } from '@nestjs/common';
import FileTypesModel, { FileTypesModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/file-types';

import { DatabaseModule } from '../../core/database/mongodb/database.module';

import { FileTypesController } from './file-types.controller';
import { FileTypesService } from './file-types.service';
import { FileTypesDal } from './file-types.dal';

@Module({
  imports    : [ DatabaseModule ],
  controllers: [ FileTypesController ],
  providers  : [
    { provide: FileTypesModelToken, useFactory: () => FileTypesModel },
    FileTypesService,
    FileTypesDal
  ],
  exports: [ FileTypesService ],
})
export class FileTypesModule {
}
