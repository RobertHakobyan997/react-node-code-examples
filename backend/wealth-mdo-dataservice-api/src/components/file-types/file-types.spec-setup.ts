import { Test, TestingModule } from '@nestjs/testing';
import { FileTypesModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/file-types';

import { mockMongooseModel } from '../../../test/mocks/mock-mongoose.model';

import { FileTypesService } from './file-types.service';
import { FileTypesDal } from './file-types.dal';

export const assert = async () => {
  const module: TestingModule = await Test.createTestingModule({
    providers: [
      FileTypesService,
      FileTypesDal,
      { provide: FileTypesModelToken, useValue: mockMongooseModel },
    ],
  }).compile();

  return module.get<FileTypesService>(FileTypesService);
};
