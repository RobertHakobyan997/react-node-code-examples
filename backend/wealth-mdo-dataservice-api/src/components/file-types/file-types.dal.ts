import { Inject, Injectable } from '@nestjs/common';
import { FilterQuery, Model, UpdateQuery } from 'mongoose';
import { Logger } from 'ngpd-merceros-logger-be';
import { plainToClass } from 'class-transformer';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import FileTypesModel, { FileTypesModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/file-types';
import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';

@Injectable()
export class FileTypesDal {
  private readonly logger = new Logger(FileTypesDal.name);

  constructor(@Inject(FileTypesModelToken) private readonly model: Model<Document<Enum>>) {
  }

  findAll(filter: FilterQuery<Document<Enum>> = {}, sort: string | any = {}, options: any = {}) {
    return this.model
      .find(filter, {}, options)
      .sort(sort)
      .exec();
  }

  findOne(filter: FilterQuery<Document<Enum>> = {}) {
    return this.model.findOne(filter).exec();
  }

  findById(id: any, options: any = {}) {
    return this.model.findById(id, {}, options).exec();
  }

  insert(docs: any[]) {
    return this.model.insertMany(plainToClass(FileTypesModel, docs));
  }

  findByIdAndUpdate(id: any, docs: UpdateQuery<Enum>) {
    return this.model.findByIdAndUpdate(id, plainToClass(FileTypesModel, docs));
  }

  findByIdAndDelete(id: any) {
    return this.model.findByIdAndDelete(id);
  }
}
