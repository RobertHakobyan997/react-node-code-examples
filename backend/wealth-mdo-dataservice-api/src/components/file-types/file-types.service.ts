import { Injectable } from '@nestjs/common';
import { FilterQuery } from 'mongoose';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';

import { FileTypesDal } from './file-types.dal';

@Injectable()
export class FileTypesService {
  constructor(private readonly dal: FileTypesDal) {
  }

  findAll(filter: FilterQuery<Document<Enum>> = {}, sort: string | any = {}, options: any = {}) {
    return this.dal.findAll(filter, sort, options);
  }
}
