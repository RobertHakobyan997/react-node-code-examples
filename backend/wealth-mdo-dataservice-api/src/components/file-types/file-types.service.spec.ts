import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import { plainToClass } from 'class-transformer';
import FileTypesModel from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/file-types';
import { Types } from 'mongoose';

import { fileTypes, fileTypesRaw } from '../../../test/data/file-types/file-types.const';

import { FileTypesService } from './file-types.service';
import { assert } from './file-types.spec-setup';

describe('FileTypesService', () => {
  let service: FileTypesService;

  beforeEach(async () => {
    service = await assert();
  });

  it('plainToClass', () => {
    const mapped = plainToClass(FileTypesModel, fileTypesRaw);
    mapped.map(el => expect(el._id).toBeInstanceOf(Types.ObjectId));
    expect(mapped).toMatchSnapshot();
  });

  describe('getFileTypes()', () => {
    it('should return all file types', async () => {
      jest.spyOn(service['dal'], 'findAll').mockResolvedValue(fileTypes as Document<Enum>[]);
      const res = await service.findAll();
      expect(res).toEqual(fileTypes);
      expect(res.length).toBe(4);
    });
  });
});
