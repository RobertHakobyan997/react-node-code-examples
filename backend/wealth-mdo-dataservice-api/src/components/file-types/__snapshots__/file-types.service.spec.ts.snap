// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileTypesService plainToClass 1`] = `
Array [
  Object {
    "_id": "609d4c33bfd3901b8c3af802",
    "display": "Positions",
    "hidden": false,
    "key": "positions",
    "order": 1,
  },
  Object {
    "_id": "609d4c33bfd3901b8c3af803",
    "display": "Transactions",
    "hidden": false,
    "key": "transactions",
    "order": 2,
  },
  Object {
    "_id": "609d4c33bfd3901b8c3af804",
    "display": "Cash Flow",
    "hidden": false,
    "key": "cashFlow",
    "order": 3,
  },
  Object {
    "_id": "609d4c33bfd3901b8c3af805",
    "display": "Error",
    "hidden": false,
    "key": "error",
    "order": 4,
  },
]
`;
